app/src/main/java/com/azuresamples/msalandroidapp/MainActivity.java

```diff
@@ -0,0 +1,552 @@
+package com.azuresamples.msalandroidapp;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.preference.PreferenceManager;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+import android.util.Pair;
+import com.android.volley.*;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+import org.json.JSONObject;
+
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+
+import com.microsoft.identity.client.*;
+import com.microsoft.identity.client.exception.*;
+
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTParser;
+
+public class MainActivity extends AppCompatActivity {
+
+    /* B2C Constants */
+    final static String B2C_SCOPES [] = {"https://fabrikamb2c.onmicrosoft.com/helloapi/demo.read"};
+
+    /* These scopes are required but not used for actual tokens */
+    final static String B2C_EDIT_SCOPES [] = {"https://fabrikamb2c.onmicrosoft.com/helloapi/demo.read"};
+    final static String B2C_RESET_SCOPES [] = {"https://fabrikamb2c.onmicrosoft.com/helloapi/demo.read"};
+
+    /* URL of the API we want to request data from */
+    final static String API_URL = "https://fabrikamb2chello.azurewebsites.net/hello";
+
+    final static String SISU_POLICY = "https://login.microsoftonline.com/tfp/fabrikamb2c.onmicrosoft.com/B2C_1_SUSI";
+    final static String EDIT_PROFILE_POLICY = "https://login.microsoftonline.com/tfp/fabrikamb2c.onmicrosoft.com/B2C_1_edit_profile";
+    final static String RESET_PASSWORD_POLICY = "https://login.microsoftonline.com/tfp/fabrikamb2c.onmicrosoft.com/B2C_1_reset";
+
+    /* UI & Debugging Variables */
+    private static final String TAG = MainActivity.class.getSimpleName();
+    Button loginButton;
```

Comment:
Totally optional : A coding style we follow as a team is to prefix member variables with "m"